### PR TITLE
[fix](load) fix stream load fail when setting strict mode

### LINKED ISSUE #12683: [Bug] stream load fail when setting strict mode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

`NumberFilteredRows` is not as expected when stream load is set to strict mode

### What You Expected?

`NumberFilteredRows` meets expectations

### How to Reproduce?

```
CREATE TABLE `test_stream_load_strict` (
  `k1` bigint(20) NULL,
  `k2` bigint(20) NULL,
  `v1` tinyint(4) SUM NULL,
  `v2` tinyint(4) REPLACE NULL,
  `v3` tinyint(4) REPLACE_IF_NOT_NULL NULL,
  `v4` smallint(6) REPLACE_IF_NOT_NULL NULL,
  `v5` int(11) REPLACE_IF_NOT_NULL NULL,
  `v6` bigint(20) REPLACE_IF_NOT_NULL NULL,
  `v7` largeint(40) REPLACE_IF_NOT_NULL NULL,
  `v8` datetime REPLACE_IF_NOT_NULL NULL,
  `v9` date REPLACE_IF_NOT_NULL NULL,
  `v10` char(10) REPLACE_IF_NOT_NULL NULL,
  `v11` varchar(6) REPLACE_IF_NOT_NULL NULL,
  `v12` decimal(27, 9) REPLACE_IF_NOT_NULL NULL
) ENGINE=OLAP
AGGREGATE KEY(`k1`, `k2`)
COMMENT 'OLAP'
PARTITION BY RANGE(`k1`)
(PARTITION partition_a VALUES [("-9223372036854775808"), ("100000")),
PARTITION partition_b VALUES [("100000"), ("1000000000")),
PARTITION partition_c VALUES [("1000000000"), ("10000000000")),
PARTITION partition_d VALUES [("10000000000"), (MAXVALUE)))
DISTRIBUTED BY HASH(`k1`, `k2`) BUCKETS 3
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"in_memory" = "false",
"storage_format" = "V2",
"disable_auto_compaction" = "false"
);

> cat replace_if_not_null_data_some
-2	-50	1	\N	j

 curl --location-trusted -u root: -H "label:sdpasdls" -H "timeout:100" -H "partitions: partition_a, partition_b, partition_c, partition_d" -H "columns: k1, k2, v2, v10, v11" -H "strict_mode: true" -H "column_separator:\t" -T replace_if_not_null_data_some http://xxxx:xxxx/api/test_stream_load_strictdb/test_stream_load_strict/_stream_load
{
    "TxnId": 7023,
    "Label": "sdpasdls",
    "TwoPhaseCommit": "false",
    "Status": "Fail",
    "Message": "too many filtered rows",
    "NumberTotalRows": 1,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 1,
    "NumberUnselectedRows": 0,
    "LoadBytes": 14,
    "LoadTimeMs": 59,
    "BeginTxnTimeMs": 1,
    "StreamLoadPutTimeMs": 4,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 50,
    "CommitAndPublishTimeMs": 0,
    "ErrorURL": "http://xxxx:xxxx/api/_load_error_log?file=__shard_0/error_log_insert_stmt_fd4d48366b34c332-72f4b5443807dd9b_fd4d48366b34c332_72f4b5443807dd9b"
}

Reason: column(v10) value is incorrect while strict mode is true, src value is . src line [-2 -50 1 NULL j]; 

```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #12683

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/base_scanner.cpp [hunk 1]
```diff
@@ -332,7 +337,8 @@ Status BaseScanner::_materialize_dest_block(vectorized::Block* dest_block) {
             for (int i = 0; i < rows; ++i) {
                 if (filter_map[i] && nullable_column->is_null_at(i)) {
                     if (_strict_mode && (_src_slot_descs_order_by_dest[dest_index]) &&
-                        !_src_block.get_by_position(dest_index).column->is_null_at(i)) {
+                        !_src_block.get_by_position(_dest_slot_to_src_slot_index[dest_index])
```
Comment (BiteTheDDDDt) [inline be/src/exec/base_scanner.cpp:340]: Better use `_dest_slot_to_src_slot_index.find` to instead `operator[]`, and add a check for not found situation

Comment (luozenglin) [reply be/src/exec/base_scanner.cpp:340]: If `_src_slot_descs_order_by_dest[dest_index]` is not nullptr, then the dest column has a src column, and `dest_index` exists in `_dest_slot_to_src_slot_index`.

File: be/src/exec/base_scanner.cpp [hunk 2]
```diff
@@ -178,6 +178,11 @@ Status BaseScanner::init_expr_ctxes() {
                 if (_src_slot_it == std::end(src_slot_desc_map)) {
                     return Status::InternalError("No src slot {} in src slot descs", it1->second);
                 }
+                auto src_slot_index = std::find(_src_slot_descs.cbegin(), _src_slot_descs.cend(),
```
Comment (BiteTheDDDDt) [inline be/src/exec/base_scanner.cpp:181]: Maybe we can maintain a map represent <slot_id -> slot_index>, to prevent find desc in vector everytime.

Comment (luozenglin) [reply be/src/exec/base_scanner.cpp]: done. thanks

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**